crates/autopilot/src/infra/solvers/mod.rs

```diff
@@ -72,11 +77,29 @@ impl Driver {
             .await
             .context("body")?;
         let text = String::from_utf8_lossy(&body);
-        tracing::trace!(%status, body=%text, "response");
+        tracing::trace!(%status, body=%text, "solver response");
         let context = || format!("url {url}, body {text:?}");
         if status != 200 {
             return Err(anyhow!("bad status {status}, {}", context()));
         }
         serde_json::from_slice(&body).with_context(|| format!("bad json {}", context()))
     }
 }
+
+/// Extracts the bytes of the response up to some size limit.
+///
+/// Returns an error if the byte limit was exceeded.
+pub async fn response_body_with_size_limit(
+    response: &mut reqwest::Response,
+    limit: usize,
+) -> Result<Vec<u8>> {
+    let mut bytes = Vec::new();
+    while let Some(chunk) = response.chunk().await? {
+        let slice: &[u8] = &chunk;
+        if bytes.len() + slice.len() > limit {
+            return Err(anyhow!("size limit exceeded"));
+        }
+        bytes.extend_from_slice(slice);
+    }
+    Ok(bytes)
+}
```

Comment:
copied from https://github.com/cowprotocol/services/blob/main/crates/shared/src/http_client.rs#L81 to break the dependency on `shared` crate.